src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java

```diff
@@ -0,0 +1,287 @@
+
+import static java.util.logging.Level.SEVERE;
+
+public class StringUtils {
+    private static Logger log = Logger.getLogger(StringUtils.class.getName());
+
+    public static void printErrMsg(String methodName, String errorMsg, int mCount, char message[]) {
+        log.log(SEVERE, methodName + ": " + errorMsg);
+        String msg1 = "ERROR: " + methodName + ": msg=\\";
+        String msg2 = "ERROR: " + methodName + ":      ";
+        for (int loopi = 0; loopi < message.length; loopi++) {
+            msg1 = msg1 + message[loopi];
+            msg2 = msg2 + " ";
+        }
+        msg1 = msg1 + "\\";
+        msg2 = msg2 + "^";
+        log.log(SEVERE, msg1);
+        log.log(SEVERE, msg1);
+    }
+
+    public static int skipWhiteSpace(int mCount, char messageChars[], String errMsg) {
+
+        while (mCount < messageChars.length) {
+            if (messageChars[mCount] == ' ' || messageChars[mCount] == '\n' || messageChars[mCount] == '\t') {
+                mCount++;
+            } else {
+                break;
+            }
+        }
+        if (errMsg != null) {
+            if (mCount >= messageChars.length) {
+                printErrMsg("RString.skipWhiteSpace", errMsg, mCount, messageChars);
+                return -1;
+            }
+        }
+        return mCount;
+    }
+
+    public static float[] parseFloatList(int numFloats, String list, int startIndex) {
+        if (list == null) {
+            return null;
+        }
+        if (list.equals("")) {
+            return null;
+        }
+
+        float[] returnArray = new float[numFloats];
+        int returnArrayCount = 0;
+
+        // Copy list into a char array.
+        char listChars[];
+        listChars = new char[list.length()];
+        list.getChars(0, list.length(), listChars, 0);
+        int listLength = listChars.length;
+
+        int count = startIndex;
+        int itemStart = startIndex;
+        int itemEnd = 0;
+        int itemLength = 0;
+
+        while (count < listLength) {
+            // Skip any leading whitespace
+            itemEnd = skipWhiteSpace(count, listChars, null);
+            count = itemEnd;
+            if (count >= listLength) {
+                break;
+            }
+            itemStart = count;
+            itemEnd = itemStart;
+            while (itemEnd < listLength) {
+                if ((listChars[itemEnd] != ' ') && (listChars[itemEnd] != '\n') && (listChars[itemEnd] != '\t')) {
+                    itemEnd++;
+                } else {
+                    break;
+                }
+            }
+            itemLength = itemEnd - itemStart;
+            returnArray[returnArrayCount++] = Float.parseFloat(new String(listChars, itemStart, itemLength));
+            if (returnArrayCount >= numFloats) {
+                break;
+            }
+
+            count = itemEnd;
+        }
+        return returnArray;
+    }
+
+    public static int[] parseIntList(String list, int startIndex) {
```

Comment:
It seems complicated to me. Maybe accept string with special delimiter, for instance, only with one whitespace. And rename it to something like stringToIntArray.